spec/models/manageiq/providers/amazon/cloud_manager/event_target_parser_spec.rb

```diff
@@ -0,0 +1,89 @@
+describe ManageIQ::Providers::Amazon::CloudManager::EventTargetParser do
+  before :each do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    @ems                 = FactoryGirl.create(:ems_amazon, :zone => zone)
+
+    allow_any_instance_of(EmsEvent).to receive(:handle_event)
+    allow(EmsEvent).to receive(:create_completed_event)
+  end
+
+  context "AWS Config Event" do
+    it "parses vm_ems_ref into event" do
+      ems_event = create_ems_event("sqs_message.json")
```

Comment:
Still, I would create an `EmsEvent` via a factory here and not test the whole EmsEvent Stream stack.

But I get your integration spec needs. But shouldnt those parts of the code already be tested somewhere else?